internal/cmd/grpc.go

```diff
@@ -494,6 +479,69 @@ func (s *GRPCServer) onShutdown(fn errFunc) {
 	s.shutdownFuncs = append(s.shutdownFuncs, fn)
 }
 
+var (
+	traceExpOnce sync.Once
+	traceExp     tracesdk.SpanExporter
+	traceExpFunc errFunc = func(context.Context) error { return nil }
+	traceExpErr  error
+)
+
+func getTraceExporter(ctx context.Context, cfg *config.Config) (tracesdk.SpanExporter, errFunc, error) {
+	traceExpOnce.Do(func() {
+		switch cfg.Tracing.Exporter {
+		case config.TracingJaeger:
+			traceExp, traceExpErr = jaeger.New(jaeger.WithAgentEndpoint(
+				jaeger.WithAgentHost(cfg.Tracing.Jaeger.Host),
+				jaeger.WithAgentPort(strconv.FormatInt(int64(cfg.Tracing.Jaeger.Port), 10)),
+			))
+		case config.TracingZipkin:
+			traceExp, traceExpErr = zipkin.New(cfg.Tracing.Zipkin.Endpoint)
+		case config.TracingOTLP:
+			u, err := url.Parse(cfg.Tracing.OTLP.Endpoint)
+			if err != nil {
+				traceExpErr = fmt.Errorf("parsing otlp endpoint: %w", err)
+				return
+			}
+
+			var client otlptrace.Client
+			switch u.Scheme {
+			case "http", "https":
+				client = otlptracehttp.NewClient(
+					otlptracehttp.WithEndpoint(u.Host+u.Path),
```

Comment:
Are we able to just do something like this?

```suggestion
					otlptracehttp.WithEndpoint(u.String()),
```

Reply:
no because `otlptrace` expects the endpoint without the scheme. `u.String()` would return the scheme (`http/https/grpc`)

Reply:
@markphelps Got it!